src/realm/object-store/c_api/sync.cpp

```diff
@@ -410,6 +411,69 @@ RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t* config,
     config->client_resync_mode = ClientResyncMode(mode);
 }
 
+RLM_API realm_object_id_t realm_flx_sync_subscription_id(const realm_flx_sync_subscription_t* subscription) noexcept
+{
+    REALM_ASSERT(subscription != nullptr);
+    return wrap_err([&]() {
+        return to_capi(subscription->id());
+    });
+}
+
+RLM_API bool realm_flx_sync_subscription_has_name(const realm_flx_sync_subscription_t* subscription) noexcept
```

Comment:
I have mostly added it for completeness

Reply:
just some background: the reason the Subscription class has a has_name() method at all is because std::string_view and Realm's StringData have a different concept for being null. It is probably sufficient to check if the return value is a nullptr_t, but it's definitely not sufficient to check the size().

Reply:
I notice that `std::string_view{}.data() == nullptr`. Given the discussion on what to return, if `has_name == false`, I think SDKs should always check `has_name` first.

Reply:
I would prefer to hide `realm_flx_sync_subscription_has_name` in order to keep the API cleaner, just return NULL if there is no name. Call internally in `realm_flx_sync_subscription_name` the method `subscrption->has_name()`

Reply:
Okay, I have removed the `_has_name` function. `_name` function will return `nullptr` if no name is set, and `\0` on an empty name.